src/test/kotlin/fr/social/gouv/agora/infrastructure/feedbackQag/repository/GetFeedbackQagRepositoryImplTest.kt

```diff
@@ -36,96 +41,147 @@ internal class GetFeedbackQagRepositoryImplTest {
+                .willReturn(CacheResult.CachedFeedbackQagNotFound)
+
+            // When
+            val result = repository.getFeedbackQagStatus(
+                qagId = qagUUID.toString(),
+                userId = userUUID.toString(),
+            )
+
+            // Then
+            assertThat(result).isEqualTo(feedbackQagStatusFalse)
+            then(cacheRepository).should(only()).getFeedbackQag(qagId = qagUUID, userId = userUUID)
+            then(databaseRepository).shouldHaveNoInteractions()
+        }
+
+        @Test
+        fun `getFeedbackQag - when CachedFeedbackQag - should return FeedbackQagDTO from cache`() {
+            // Given
+            val qagUUID = UUID.randomUUID()
+            val userUUID = UUID.randomUUID()
+            given(cacheRepository.getFeedbackQag(qagId = qagUUID, userId = userUUID))
+                .willReturn(CacheResult.CachedFeedbackQag(feedbackQagDTO))
+
+            // When
+            val result = repository.getFeedbackQagStatus(
+                qagId = qagUUID.toString(),
+                userId = userUUID.toString(),
+            )
+
+            // Then
+            assertThat(result).isEqualTo(feedbackQagStatusTrue)
+            then(cacheRepository).should(only()).getFeedbackQag(qagId = qagUUID, userId = userUUID)
+            then(databaseRepository).shouldHaveNoInteractions()
+        }
     }
 
-    @Test
-    fun `getFeedbackQag - when invalid userId - should return null without doing anything`() {
-        // When
-        val result = repository.getFeedbackQagStatus(
-            qagId = UUID.randomUUID().toString(),
-            userId = "invalid userId",
-        )
-
-        // Then
-        assertThat(result).isNull()
-        then(cacheRepository).shouldHaveNoInteractions()
-        then(databaseRepository).shouldHaveNoInteractions()
+    @Nested
+    inner class GetFeedbackQagListTestCases {
+        @Test
+        fun `getFeedbackQagList - when invalid qagId - should return emptyList() without doing anything`() {
+            // When
+            val result = repository.getFeedbackQagList(
+                qagId = "invalid qagId UUID",
+            )
+
+            // Then
+            assertThat(result).isEqualTo(emptyList<FeedbackQagInserting>())
+            then(databaseRepository).shouldHaveNoInteractions()
+        }
+
+        @Test
+        fun `getFeedbackQagList - when valid qagId and database return emptyList() - should return emptyList()`() {
+            // Given
+            val qagUUID = UUID.randomUUID()
+            given(databaseRepository.getFeedbackQagList(qagId = qagUUID)).willReturn(emptyList())
+
+            // When
+            val result = repository.getFeedbackQagList(
+                qagId = qagUUID.toString(),
+            )
+
+            // Then
+            assertThat(result).isEqualTo(emptyList<FeedbackQagInserting>())
+            then(databaseRepository).should(only()).getFeedbackQagList(qagId = qagUUID)
```

Comment:
Pas obligé mais on peut faire l'ajout d'un `then(mapper).should...` ici

Reply:
done